src/InputStream.js

```diff
@@ -25,7 +37,13 @@ InputStream.prototype = {
     return this.source[this.pos++];
   },
 
-  matchString: function(s) {
+  matchString: function(s, optIgnoreCase) {
+    if (optIgnoreCase) {
+      var substring = this.sourceSlice(this.pos, this.pos + s.length);
+      var match = regExpForString(s, IGNORE_CASE).exec(substring);
```

Comment:
Oh yes, I didn't think about the number of characters in the string -- yay FTW :) But in that case, the RegExp thing wouldn't have worked anyway, because `substring` was computed from `s.length`, right?

And about the caching thing: there are cases where you *would* be able to cache the lower-cased string in the application pexpr -- I'm thinking of when the argument to `CaseInsensitive` is a string literal pexpr (so you know it's always going to be the same thing). But in general, you're right... it could be a variable (one of the formal arguments of the rule from which `CaseInsensitive` is being called) and in that case we wouldn't be able to cache it.